### DISCUSSION TITLE
Renovate is using the wrong PHP version

### DISCUSSION THREAD

**Discussion by clemlatz** [renovatebot/renovate#31670] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi!

I want my open source project to support all currently supported PHP versions (that's 8.1 to 8.3 as I write) so I'd like to prevent Renovate to propose packages updates that are requiring at least PHP 8.2.

My [composer.json](https://github.com/biblys/biblys/blob/de39d5ee8b614b53f2526b8f5e6b91c01bfa2880/composer.json) file states:

```json
"require": {
    "php": "~8.1",
```

And my [renovate.json](https://github.com/biblys/biblys/blob/dev/renovate.json) states:

```json
"composer": {
    "constraints": {
      "php": "8.1.29"
    }
  },
```

Still, [renovate opens PRs](https://github.com/biblys/biblys/pull/63) where [the CI fails when tests are executed using PHP 8.1](https://github.com/biblys/biblys/actions/runs/10962529780/job/30441988335?pr=63) because Renovate selects composer packages versions that requires at least PHP 8.2. Eg for phpunit, I expect version 10 which is still compatible with PHP 8.1, but:

```
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - phpunit/phpunit[11.0.0, ..., 11.3.6] require php >=8.2 -> your php version (8.1.29) does not satisfy that requirement.
    - Root composer.json requires phpunit/phpunit ~11 -> satisfiable by phpunit/phpunit[11.0.0, ..., 11.3.6].

Error: Your requirements could not be resolved to an installable set of packages.
```

Am I doing something wrong? Is there a way to force Renovate to respect version constraints (ideally the latest 8.1 patch and not a specific 8.1.x patch)?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Look up "constraintsFiltering"
